clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2355,6 +2492,15 @@ Applies to a device function/lambda function or function call operator (of a
 function object). If 1, compiler doesn't use the global work offset values for
 the device function. Valid values are 0 and 1. If used without argument, value
 of 1 is set implicitly.
+
+.. code-block:: c++
+
+  [[intel::no_global_work_offset(N)]] int foo() {}
+
+``[[intel::no_global_work_offset(N)]]`` takes one optional parameter that is a
+constant unsigned integer expression. The parameter N may be a template
```

Comment:
I'm on the fence about "The parameter N may be a template parameter" -- the sentence isn't accurate as it is because that implies you can do:
```
template <typename ThisIsATemplateParameter>
[[intel::no_global_work_offset(ThisIsATemplateParameter)]] void whatever() {}
```
However, if you say it can be a non-type template parameter, that's redundant with the first sentence in the paragraph. I sort of think we can drop the sentence entirely and rely on an example to get the point across that a non-type template parameter is also a constant expression.

Reply:
Added examples and removed the sentence entirely.